include/rpm/rpmfileutil.h

```diff
@@ -110,11 +118,13 @@ char * rpmGetPath (const char * path, ...) RPM_GNUC_NULL_TERMINATED;
 /** \ingroup rpmfileutil
  * Return URL path(s) from a (URL prefixed) pattern glob.
  * @param pattern	glob pattern
+ * @param flags		glob flags
  * @param[out] *argcPtr	no. of paths
  * @param[out] *argvPtr	ARGV_t array of paths
  * @return		0 on success
  */
-int rpmGlob(const char * pattern, int * argcPtr, ARGV_t * argvPtr);
+int rpmGlob(const char * pattern, rpmglobFlags flags,
+	    int * argcPtr, ARGV_t * argvPtr);
```

Comment:
Not a complete review but this caught my eye:

Technically, this is part of our public API and we can't just silently change it. 

The normal way to handle this kind of thing is to add a new function (rpmGlob2() or similar, ugly as it may be) with the flags argument, and make the original call that with flags that equal to prior behavior. That in a commit of it's own (this is good for proving the original interface still works as before) and only then update callers to use the new API where needed - only the rpmGlob() calls in build/files.c need changing then.

Reply:
Oh, I was counting on the fact that we'd have bumped the soname whenever this is released, but in this case there's a way to do this simply by adding another function. I'll do that, thanks!

Reply:
Yup, there's the soname bump of course, which would allow us to do this, but it should be mentioned in the commit message. In general, preserving API where possible is good, but rpmGlob() is one of those cases where you wonder if anything at all outside rpm itself is using it, ie is it worth the trouble at all.

Reply:
True, and yes, this one a "meh" case perhaps, but still, let's just go with another function. We can always change our mind in the future.